config/migrate.ts

```diff
@@ -0,0 +1,191 @@
+}
+
+export function deleteConfigFile(useHiddenConfig = false): void {
+  if (useHiddenConfig) {
+    newDeleteConfigFile();
+  }
+  config_DEPRECATED.deleteConfigFile();
+}
+
+function writeGlobalConfigFile(updatedConfig: CLIConfig_NEW): void {
+  const updatedConfigJson = JSON.stringify(updatedConfig);
+  createEmptyConfigFile({}, true);
+  loadConfig('');
+
+  try {
+    writeConfig({ source: updatedConfigJson });
+    deleteConfigFile();
+  } catch (error) {
+    deleteEmptyConfigFile();
+
+    throw new Error(
+      i18n(`${i18nKey}.errors.writeConfig`, { configPath: GLOBAL_CONFIG_PATH }),
+      { cause: error }
+    );
+  }
+}
+
+export function migrateConfig(
+  deprecatedConfig: CLIConfig_DEPRECATED | null
+): void {
+  if (!deprecatedConfig) {
+    throw new Error(i18n(`${i18nKey}.errors.noDeprecatedConfig`));
+  }
+  const { defaultPortal, portals, ...rest } = deprecatedConfig;
+  const updatedConfig = {
+    ...rest,
+    defaultAccount: defaultPortal,
+    accounts: portals
+      .filter(({ portalId }) => portalId !== undefined)
+      .map(({ portalId, ...rest }) => ({
+        ...rest,
+        accountId: portalId!,
+      })),
+  };
+  writeGlobalConfigFile(updatedConfig);
+}
+
+function mergeConfigPropertes(
+  globalConfig: CLIConfig_NEW,
+  deprecatedConfig: CLIConfig_DEPRECATED
+): CLIConfig_NEW {
+  const propertiesToCheck: Array<keyof Partial<CLIConfig>> = [
+    'defaultCmsPublishMode',
+    'httpTimeout',
+    'allowUsageTracking',
+    'env',
+  ];
+  const conflicts: Array<{
+    property: keyof CLIConfig_NEW;
+    oldValue: boolean | string | number | CmsPublishMode | Environment;
+    newValue: boolean | string | number | CmsPublishMode | Environment;
+  }> = [];
+
+  propertiesToCheck.forEach(prop => {
+    if (prop in globalConfig && prop in deprecatedConfig) {
+      if (globalConfig[prop] !== deprecatedConfig[prop]) {
+        conflicts.push({
+          property: prop,
+          oldValue: deprecatedConfig[prop]!,
+          newValue: globalConfig[prop]!,
+        });
+      }
+    } else {
+      // @ts-expect-error TODO
+      globalConfig[prop] = deprecatedConfig[prop];
+    }
+  });
+
+  if ('defaultAccount' in globalConfig && 'defaultPortal' in deprecatedConfig) {
+    if (globalConfig.defaultAccount !== deprecatedConfig.defaultPortal) {
+      conflicts.push({
+        property: 'defaultAccount',
+        oldValue: deprecatedConfig.defaultPortal!,
+        newValue: globalConfig.defaultAccount!,
+      });
+    }
+  } else if ('defaultPortal' in deprecatedConfig) {
+    globalConfig.defaultAccount = deprecatedConfig.defaultPortal;
+  }
+
+  if (conflicts.length > 0) {
+    logger.log(
+      `The following properties have different values in the deprecated and global config files:\n${conflicts
+        .map(
+          ({ property, oldValue, newValue }) =>
+            `${property}: ${oldValue} (deprecated) vs ${newValue} (global)`
+        )
+        .join('\n')}`
+    );
+    return globalConfig;
```

Comment:
Nit: This return statement is redundant because we have another return as the next statement below it.